### PR TITLE
Post review

### PR DESCRIPTION
**Pull request by rjucf**
Renamed Promo to Spotlight.
Updated CSS so that it is not specific for ucf.edu and pulled in the Athena btn class.
Added setting to not include css.
Added the layout to the spotlight post type and removed it from the shortcode.
Linked the shortcode to the slug instead of the title.

### CODE REVIEW

**Comment by rjucf** [comment]
Which functions are referring to Jim? The functions for displaying the layouts on the page are in common.php file. Are you referring to the functions to display meta boxes?

File: includes/ucf-spotlight-common.php [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+			}
+		}
+
+		public static function display_spotlight( $attr ) {
+			$output = ucf_spotlight_display( $attr );
+			return apply_filters( 'ucf_spotlight_display', $output );
+		}
+	}
+
+	add_action( 'wp_enqueue_scripts', array( 'UCF_Spotlight_Common', 'enqueue_styles' ), 99 );
```
Comment (cjg89) [inline includes/ucf-spotlight-common.php]: Any particular reason for using 99 as the priority level?  I think this _might_ pose problems for folks using the plugin who want to enqueue an additional stylesheet before or after the plugin's CSS by setting 'ucf_spotlight_css' as a dependent stylesheet--unless they can use an action priority greater than 99, I don't think they'll be able to time it correctly.

File: includes/ucf-spotlight-common.php [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+				'header'    => get_post_meta( $post->ID, "ucf_spotlight_header", True ),
+				'copy'      => get_post_meta( $post->ID, "ucf_spotlight_copy", True ),
+				'link_text' => get_post_meta( $post->ID, "ucf_spotlight_link_text", True ),
+				'link_url'  => get_post_meta( $post->ID, "ucf_spotlight_link_url", True ),
+				'image'     => $spotlight_image,
+			);
+
+			switch ( $args['layout'] ) {
+				case "horizontal":
+					return ucf_spotlight_horizontal( $args );
```
Comment (cjg89) [inline includes/ucf-spotlight-common.php]: I think this is what Jim was referring to earlier--but consider making these functions (ucf_spotlight_horizontal/vertical/square()) methods of the UCF_Spotlight_Common class.

File: includes/ucf-spotlight-common.php [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+				case "vertical":
+					return ucf_spotlight_vertical( $args );
+				case "square":
+					return ucf_spotlight_square( $args );
+				default:
```
Comment (cjg89) [inline includes/ucf-spotlight-common.php]: Just for future reference, you could remove line 61 here (use an empty statement for 'square', as a shorthand) and the same logic would apply.

File: includes/ucf-spotlight-common.php [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+* @return String
+**/
+if ( ! function_exists( 'ucf_spotlight_horizontal' ) ) {
+	function ucf_spotlight_horizontal( $args ) {
+		ob_start();
+	?>
+				</div>
+			</div>
+		</div>
+		<section class="spotlight-horizontal" style="background-image: url(<?php echo $args['image'] ?>)">
```
Comment (cjg89) [inline includes/ucf-spotlight-common.php]: Consider creating and pulling multiple image sizes (using custom thumbnails) to use at different screen widths to avoid loading large background images on mobile devices--or, remove background images entirely at mobile sizes, so we don't even have to think about making background images scale down for these.

File: src/scss/ucf-spotlight.scss [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  }
+
+  .btn {
+    width: 213px;
+  }
+
+  .spotlight-btn-wrapper {
+    bottom: 30px;
+    left: 50%;
+    margin-left: -107px;
```
Comment (cjg89) [inline src/scss/ucf-spotlight.scss]: Avoid fixed negative margins like this for centering absolutely-positioned elements--remove the margin-left and add `transform: translateX(-50%)` instead for more accurate centering.

File: src/scss/ucf-spotlight.scss [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    margin-bottom: 15px;
+    text-shadow: 1px 1px 1px #000;
+  }
+
+  .spotlight-copy {
+    line-height: 24px;
+  }
+
+  .btn {
+    width: 213px;
```
Comment (cjg89) [inline src/scss/ucf-spotlight.scss]: Remove this fixed width (see notes on .spotlight-btn-wrapper margin-left).  If you want to make sure the button spans a minimum amount of space, use min-width instead.

File: src/scss/ucf-spotlight.scss [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+  }
+
+  .btn-primary {
+    @include button-variant($btn-primary-color, $btn-primary-bg, $btn-primary-border);
+  }
+
+}
+
+// Horizontal spotlight layout
+.spotlight-horizontal {
```
Comment (cjg89) [inline src/scss/ucf-spotlight.scss]: Horizontal spotlights don't have background-size, position, or repeat rules assigned to them

File: src/scss/mixins/_font-family.scss [hunk 8]
```diff
@@ -0,0 +1,104 @@
+//
```
Comment (cjg89) [inline src/scss/mixins/_font-family.scss]: Are the mixins included in this file necessary?  They don't appear to be used anywhere in the plugin css

File: src/scss/mixins/_letter-spacing.scss [hunk 9]
```diff
@@ -0,0 +1,10 @@
+//
```
Comment (cjg89) [inline src/scss/mixins/_letter-spacing.scss]: Is this file necessary?

File: static/css/ucf-promo.min.css [hunk 10]
```diff
@@ -1 +0,0 @@
...promo-square .promo-header,.promo-square .promo-link{color:#fff;text-align:center}.promo-square .promo-header{font-size:26px!important;margin-bottom:15px;text-shadow:2px 2px 3px #666}.promo-square .promo-copy{font-size:16px}.promo-square .promo-btn{transition:background-color .4s,color .4s;width:213px}.promo-square .promo-btn:hover,.promo-square:hover .promo-btn{background:#000;color:#fff;text-decoration:none;transition:background-color .4s,color .4s}.promo-square .promo-btn-wrapper{bottom:30px;left:50%;margin-left:-107px;position:absolute;text-align:center}.promo-vertical .promo-image{max-width:100%}.promo-vertical .promo-copy,.promo-vertical .promo-header{color:#000}.promo-vertical .promo-header{font-size:26px!important;margin:40px 0 15px}.promo-vertical .promo-copy{font-size:16px}.promo-vertical .promo-copy .promo-link{border-bottom:0!important;color:#337ab7}.promo-vertical .promo-copy .promo-link:hover{text-decoration:underline!important}.promo-vertical .promo-btn{border-bottom:0!important;transition:background-color .4s,color .4s}.promo-vertical .promo-btn:hover{background:#000;border-bottom:0!important;color:#fff;text-decoration:none;transition:background-color .4s,color .4s}
```
Comment (cjg89) [inline static/css/ucf-promo.min.css]: This file needs to be removed

File: src/scss/mixins/_bootstrap.scss [hunk 11]
```diff
@@ -0,0 +1,100 @@
+@mixin border-radius($radius: $border-radius) {
```
Comment (cjg89) [inline src/scss/mixins/_bootstrap.scss]: Would it maybe be easier to just copy the compiled css for .btn's into the plugin instead of copying down multiple mixin/variable files from Athena?

File: static/css/ucf-promo.min.css (hunk 10 above)
Comment (cjg89) [inline static/css/ucf-promo.min.css]: d'oh...nevermind

**Comment by rjucf** [comment]
Code Review Fixes: https://github.com/UCF/UCF-Spotlights-Plugin/compare/62efe00750d0730221eb2267e4ebc0eb9d20d1bc...post-review?w=1

File: includes/ucf-spotlight-common.php [hunk 12]
```diff
@@ -13,162 +13,152 @@ public function enqueue_styles() {
...
-		// get post with $slug
-		if( !empty( $attr['slug'] ) ) {
+		/**
+		* Returns the spotlight HTML to be displayed on the page.
+		* @author RJ Bruneel
+		* @since 1.0.0
+		* @param $attr string | slug of the spotlight post type and one of three layouts.
+		* @return String
+		**/
+		function ucf_spotlight_display( $attr ) {
```
Comment (cjg89) [inline includes/ucf-spotlight-common.php]: These methods (`ucf_spotlight_X()`) should be defined as static functions

Comment (rjucf) [inline includes/ucf-spotlight-common.php]: Updated, I set them to private as well. https://github.com/UCF/UCF-Spotlights-Plugin/commit/6badba4784cc228e0e4ec6ce4c80c49f25dd299a

File: includes/ucf-spotlight-common.php (hunk 2 above)
Comment (jmbarne3) [inline includes/ucf-spotlight-common.php]: These are the functions I was referring to.
